base/loading.jl

```diff
@@ -295,16 +297,17 @@ then tries paths in the global array `LOAD_PATH`. `require` is case-sensitive on
 all platforms, including those with case-insensitive filesystems like macOS and
 Windows.
 """
-function require(mod::Symbol)
+function require(from::Module, mod::Symbol)
     if !root_module_exists(mod)
```

Comment:
I don't know if this check is still sensible. It depends on what it means to require a root module from within an other module.
Currently we are using root-modules for the `stdlib` packages, since we don't want to make them available under `Main` immediately. As I noted on Slack this means that we can't update stdlib packages that are backed into the sysimage.